File: PersonalLifeOS.Domain/Entities/Bill.cs
namespace PersonalLifeOS.Domain.Entities;

public class Bill
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public decimal Amount { get; set; }
    public DateTime DueDate { get; set; }
    public int ReminderDaysBefore { get; set; } = 3;
    public string StatusCode { get; set; } = "Pending";
    public string UserId { get; set; } = string.Empty;

    // Audit fields
    public string CreatedBy { get; set; } = string.Empty;
    public DateTime CreatedDate { get; set; }
    public string? UpdatedBy { get; set; }
    public DateTime? UpdatedDate { get; set; }
}

File: frontend/src/types/bill.ts
export interface Bill {
  id: number;
  name: string;
  amount: number;
  dueDate: string;
  statusCode: string;
  reminderDaysBefore: number;
}

export interface CreateBillRequest {
  name: string;
  amount: number;
  dueDate: string;
  reminderDaysBefore: number;
}

export interface UpdateBillRequest extends CreateBillRequest {
  id: number;
  statusCode: string;
}
